File: python-version/LC1706/find_ball.py
class Solution:
    def findBall(self, grid: list[list[int]]) -> list[int]:
        n = len(grid[0])
        ans = [0] * n
        for i in range(n):
            curCol = i
            for row in grid:
                dir = row[curCol]
                nextCol = curCol + dir
                if nextCol < 0 or nextCol >= n or row[nextCol] != dir:
                    curCol = -1
                    break
                curCol = nextCol
            ans[i] = curCol
        return ans

File: go-version/leetcode/LC1706/find_ball.go
package lc1706

func findBall(grid [][]int) []int {
	n := len(grid[0])
	ans := make([]int, n)
	for i := 0; i < n; i++ {
		curCol := i
		for _, row := range grid {
			dir, nextCol := row[i], curCol+row[i]
			if nextCol < 0 || nextCol >= n || row[nextCol] != dir {
				curCol = -1
				break
			}
			curCol = nextCol
		}
		ans[i] = curCol
	}
	return ans
}
